### PR TITLE
cache the npm cache on jenkins workers

### PR DESCRIPTION
**Pull request by estute**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
----

Similar to how we archive the python virtualenv for a dependency baseline, archive the NPM local cache.

### CODE REVIEW

**Comment by estute** [comment]
Worker built here: https://test-jenkins.testeng.edx.org/job/build-packer-ami/8331/
Testing on Test Jenkins w/ PR: https://github.com/edx/edx-platform/pull/18959

File: playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml [hunk 1]
```diff
@@ -36,6 +36,29 @@
     chdir: "{{ jenkins_home }}"
   become_user: "{{ jenkins_user }}"
 
+- name: Add script to set install node packages
+  template:
+    src=jenkins_nodeenv.j2 dest={{ jenkins_home }}/jenkins_nodeenv
+    owner={{ jenkins_user }} group={{ jenkins_group }} mode=0500
```
Comment (jmbowman) [inline playbooks/roles/jenkins_worker/tasks/python_platform_worker.yml]: I realize we're using this syntax in a few other places also, but our Ansible code conventions specifically discourage it: https://openedx.atlassian.net/wiki/spaces/OpenOPS/pages/26837527/Ansible+Code+Conventions

File: playbooks/roles/jenkins_worker/templates/jenkins_nodeenv.j2 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+cd {{ jenkins_home }}/shallow-clone
```
Comment (jmbowman) [inline playbooks/roles/jenkins_worker/templates/jenkins_nodeenv.j2]: Maybe add a comment above this summarizing what the script does?

**Review by jmbowman — COMMENTED** [review]
Looks pretty good, just a couple of minor nits.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**